languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -172,6 +172,140 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+                </pattern>
+                <message>Possível erro de digitação.</message>
+                <suggestion>caso</suggestion>
+                <example correction="Caso"><marker>Case</marker> eles não venham...</example>
+                <example correction="Caso"><marker>Case</marker> os outros caras precisem de algo...</example>
+                <example correction="Caso"><marker>Case</marker> boas oportunidades apareçam...</example>
+                <example correction="Caso"><marker>Case</marker> ganhem, vão querer um prêmio.</example>
+                <example correction="Caso"><marker>Case</marker> quarenta ladrões roubem o tesouro...</example>
+                <example>Case esse casal, divorcie aquele!</example>
+                <example>Case esses dois homens e não discuta com a lei!</example>
+            </rule>
+        </rulegroup>
+
+        <rulegroup id="OUVE_HOUVE_CONFUSAO" name="Confusão entre 'ouve' e 'houve'" default="temp_off">
+            <rule id="O_QUE_OUVE">
+                <pattern>
+                    <token postag="SENT_START"/>
+                    <token min="0">o</token>
+                    <token regexp="yes">qu[eê]</token>
+                    <marker>
+                        <token>ouve</token>
+                    </marker>
+                </pattern>
+                <message>No sentido de &quot;acontecer&quot;, o verbo usado é &quot;haver&quot;.</message>
+                <suggestion>houve</suggestion>
+                <example correction="houve">O que <marker>ouve</marker>?</example>
+                <example correction="houve">O que <marker>ouve</marker> com ela?</example>
+            </rule>
+
+            <rule id="OUVE_ALGO">
+                <pattern>
+                    <marker>
+                        <token>ouve
+                            <exception scope="previous" postag_regexp="yes" postag="P[PDI].+"/>
+                            <exception scope="previous" postag_regexp="yes" postag="[NA].+"/>
+                        </token>
+                    </marker>
+                    <token>algo</token>
+                    <token min="0" postag_regexp="yes" postag="AQ.MS."/>
+                    <token min="0" regexp="yes">com(igo)?|con([st]igo|vosco|n?osco)</token>
+                </pattern>
+                <message>No sentido de &quot;acontecer&quot;, o verbo usado é &quot;haver&quot;.</message>
+                <suggestion>houve</suggestion>
+                <example correction="Houve"><marker>Ouve</marker> algo com ele?</example>
+                <example correction="houve">Será que <marker>ouve</marker> algo errado com alguém?</example>
+                <example correction="houve">Você acha que <marker>ouve</marker> algo estranho conosco?</example>
+                <example>Você ouve algo?</example>
+                <example>Alguém ouve algo?</example>
+                <example>O cachorro ouve algo ao longe.</example>
+            </rule>
```

Comment:
I suppose you're right, but if both the complementiser 'que' and the coordinating conjunction 'e' can introduce valid 'ouvir' cases, I suppose we need an even more restrictive context...

Reply:
At any rate, the specific examples you've provided don't match (just tested them) because of the two tokens after `algo`, which can't just be anything, but must necessarily be something like 'algo [estranho] [com]...'

The one case that needs to be added is stuff like 'ela para e ouve algo'.